analyzers/tests/SonarAnalyzer.Test/TestCases/LoggersShouldBeNamedForEnclosingType.cs

```diff
@@ -0,0 +1,161 @@
+﻿using Microsoft.Extensions.Logging;
+using System;
+
+class AnotherType : EnclosingType { }
+
+class EnclosingType
```

Comment:
Add test case for the following:
- aliased type: `using AliasedFactory = Microsoft.Extensions.Logging.ILoggerFactory;`
- `using static` (e.g. for the `LogManager` in `NLog`)
- constant string value: private const string ConstLoggerName = "AnotherType"; (you can leave it as FN or implement it by retrieving the `ConstantValue` from `IFieldSymbol`)